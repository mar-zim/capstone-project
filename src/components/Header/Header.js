import React from 'react'
import { useHistory } from 'react-router-dom'
import styled from 'styled-components'

export default function Header() {
  const history = useHistory()

  return (
    <StyledHeader>
      <StyledLogo
        src={process.env.PUBLIC_URL + '/logo.svg'}
        alt="logo"
        onClick={goToHome}
      />
    </StyledHeader>
  )

  function goToHome() {
    let path = `/home`
    history.push(path)
  }
}

const StyledHeader = styled.header`
  background-color: var(--grey-5);
  display: grid;
  align-items: center;
  justify-items: center;
`

const StyledLogo = styled.img`
  max-width: 80vw;
  cursor: pointer;
`
