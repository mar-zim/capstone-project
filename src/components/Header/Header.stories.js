import React from 'react'
import Header from './Header'
import { addDecorator } from '@storybook/react'

addDecorator((storyFn) => <div style={{ width: '375px' }}>{storyFn()}</div>)

export default {
  component: Header,
  title: 'General/Header',
}

export const Default = () => <Header />
