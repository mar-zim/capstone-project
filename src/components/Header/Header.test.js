import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import Header from './Header'
import pretty from 'pretty'

describe('Header-Test', () => {
  it('should render the logo as an image', () => {
    const { queryByAltText } = render(<Header />, { wrapper: MemoryRouter })
    expect(queryByAltText('logo').tagName).toBe('IMG')
  })

  it('matches snapshot', () => {
    const { container } = render(<Header />, { wrapper: MemoryRouter })
    expect(pretty(container.innerHTML)).toMatchSnapshot()
  })
})
