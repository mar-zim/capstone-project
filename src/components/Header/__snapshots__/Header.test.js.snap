// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header-Test matches snapshot 1`] = `"<header class=\\"sc-AxjAm bFPmOx\\"><img src=\\"/logo.svg\\" alt=\\"logo\\" class=\\"sc-AxirZ ctqnPd\\"></header>"`;
